src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+    minim = d.σ > 0 ? minimum(d.ρ) : maximum(d.ρ)
+    return d.μ + d.σ * minim
+end
+
+function extrema(d::UnivariateAffine)
+    extremes = extrema(d.ρ)
+    extremes = _flip(d.σ, extremes)
+    return @. d.μ + d.σ * extremes
+end
+
+
+## Multivariate
+# when finding the extrema, we need to flip the extremes if the scale is negative
+_flip(signed::Real, tuple::Tuple) = signed > 0 ? tuple : (last(tuple), first(tuple))
+
+function minimum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    # The minimum of d.ρ[i] will minimize d[j] if d.σ[i,j] is negative. 
+    # Otherwise the maximum will minimize d[j]
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    return mins
+end
+
+function maximum(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return maxes
+end
+
+function extrema(d::MultivariateAffine)
+    extremes = extrema(d.ρ) |> zip |> collect |> permutedims
+    extreme_matrix = _flip.(d.σ, extremes)
+    argmins = first.(extreme_matrix)
+    mins = [σ ⋅ argmin for (σ, argmin) in zip(eachrow(d.σ), eachrow(argmins))]
+    argmaxes = last.(extreme_matrix)
+    maxes = [σ ⋅ argmax for (σ, argmax) in zip(eachrow(d.σ), eachrow(argmaxes))]
+    return (mins, maxes)
+end
+
+support(d::AffineDistribution) = affine_support(d.μ, d.σ, support(d.ρ))
+affine_support(μ::Real, σ::Real, support) = μ .+ σ .* support
+
+function affine_support(μ::Real, σ::Real, support::RealInterval) 
+    if σ > 0
+        lower = support.lb
+        upper = support.ub
+    else
+        lower = support.ub
+        upper = support.lb
+    end
+    return RealInterval(μ + σ * lower, μ + σ * upper)
+end
+
+
+#### Conversions
+
+function convert(::Type{AffineDistribution{F,S,Tμ,Tσ,D}}, d::AffineDistribution) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    return AffineDistribution(Tμ(d.μ), Tσ(d.σ), convert(D, d.ρ))
+end
+
+
+function Base.eltype(::Type{<:AffineDistribution{F,S,Tμ,Tσ,D}}) where {
+        F<:VariateForm,
+        S<:ValueSupport,
+        D<:Distribution{F, S},
+        Tμ<:ArrayLike,
+        Tσ<:ArrayLike
+    }
+    T = Core.Compiler.return_type(Base.:*, (Tσ, eltype(D)))
+    return Core.Compiler.return_type(Base.:+, (Tμ, T))
+end
+
+
+#### Parameters
+
+function location(d::AffineDistribution)
+    if hasmethod(location, (typeof(d.ρ),))
+        return d.μ + d.σ * location(d.ρ)
+    else
+        return d.μ
+    end
+end
+function scale(d::AffineDistribution)
+    if hasmethod(scale, (typeof(d.ρ),))
+        return d.σ * scale(d.ρ)
+    else
+        return d.σ
+    end
+end
```

Comment:
As @mschauer said, this should just be
```suggestion
location(d::AffineDistribution) = d.μ + d.σ * location(d.ρ)
scale(d::AffineDistribution) = d.σ * scale(d.ρ)
```

Reply:
In cases where a distribution is made into a location-scale type family by affine transformation, the shift and scale values of the affine transformation are usually considered to be the corresponding location and scale parameters. For instance, the t distribution is usually defined as having one parameter, ν, but if I asked you for the location parameter of a Student t distribution, it would be reasonable to assume I was looking for the shift parameter of an affinely-transformed Student t distribution.

Regardless of whether we consider these to be "True" location and scale parameters, the important question is what the user is likely to want when they call `location` and `scale` on an affinely-shifted distribution. I suspect users getting an error message when they call `location(2*TDist(x) + 1)` will probably be frustrated or confused that this does not return `1` the same way that calling `location(2*Normal() + 1)` does.

In addition, removing `location` and `scale` functions for affine transformations would be a breaking change, as they are currently defined for `LocationScale`-type families.